artwork/Modules/InventoryManagement/Services/CraftInventoryItemService.php

```diff
@@ -8,44 +8,38 @@
 use Artwork\Modules\InventoryManagement\Repositories\CraftsInventoryColumnRepository;
 use Throwable;
 
-readonly class CraftInventoryItemService
+class CraftInventoryItemService
 {
     public function __construct(
-        private CraftsInventoryColumnRepository $craftsInventoryColumnRepository,
-        private CraftInventoryItemRepository $craftInventoryItemRepository,
-        private CraftInventoryItemCellService $craftInventoryItemCellService,
-        private InventoryResourceCalculateModelsOrderService $inventoryResourceCalculateModelsOrderService
+        private readonly CraftsInventoryColumnRepository $craftsInventoryColumnRepository,
+        private readonly CraftInventoryItemRepository $craftInventoryItemRepository,
+        private readonly CraftInventoryItemCellService $craftInventoryItemCellService,
+        private readonly InventoryResourceCalculateModelsOrderService $inventoryResourceCalculateModelsOrderService
     ) {
     }
 
-    public function getNewCraftInventoryItem(array $attributes): CraftInventoryItem
-    {
-        return new CraftInventoryItem($attributes);
-    }
-
     /**
      * @throws Throwable
      */
     public function create(
         int $groupId,
         int $order
     ): CraftInventoryItem {
-        $craftInventoryItem = $this->getNewCraftInventoryItem(
+        $craftInventoryItem = $this->craftInventoryItemRepository->getNewModelInstance(
             [
                 'craft_inventory_group_id' => $groupId,
                 'order' => $order
             ]
         );
         $this->craftInventoryItemRepository->saveOrFail($craftInventoryItem);
 
-        $this->craftsInventoryColumnRepository->getAllOrdered()->each(
-            function (CraftsInventoryColumn $column) use ($craftInventoryItem): void {
-                $this->craftInventoryItemCellService->create(
-                    $column->id,
-                    $craftInventoryItem->id
-                );
-            }
-        );
+        /** @var CraftsInventoryColumn $column */
+        foreach ($this->craftsInventoryColumnRepository->getAllOrdered() as $column) {
```

Comment:
`->each` sollte besser für den RAM sein

Reply:
Lässt sich nur kompliziert testen. Performance ist hier aber 1a, da liegen nicht viele Daten hinter.

Dann muss man zwei Tests schreiben, einer der das each expected und einer mit ner richtigen Collection wo man den Callback testet. Wenn man dann noch auf n Service ausm constructor zugreift hat man den Salat.

Reply:
Es spricht ja nichts dagegen gegen die DB zu testen. In den meisten Fällen ist das sogar gut so, weil du so gleich sicherstellen kannst, dass es richtig in die db geschrieben und gelesen wird.

Reply:
Tut es nicht, das macht man aber im Feature Test (die hier nicht geschrieben wurden). Unit Test werden üblicherweise nicht im Applikations-Kontext ausgeführt. Daher gibt es keine DB/keine Connections/keine Configs.

Unit -> Feature -> Application (-Test)